components/ca-form.js

```diff
@@ -0,0 +1,1520 @@
+         * @param {string} key - item object key
+         * @param {object} schema - the form schema
+         * @returns {object} dom object
+         */
+        createListBuilder(type, key, schema) {
+
+            let el = null;
+
+            if (schema.type === 'array' && schema.items && (schema.items.extends || schema.items.links)) {
+                el = createElementLegacy(null, 'ca-list-builder', { name: key, id: key, layout: 'sidebyside' });
+                el.schema = schema.items;
+            }
+
+            return el;
+        }
+
+        /**
+         * @description create a ca-remote-image-capture
+         * @param {string} type - type of object
+         * @param {string} key - item object key
+         * @param {object} schema - the form schema
+         * @returns {object} dom object
+         */
+        createImageCapture(type, key, schema) {
+
+            let el = null;
+
+            if (schema.media && schema.media.type === 'image/jpeg' && schema.media.binaryEncoding === 'base64') {
+                // TODO: tech debt, user mapper or similar instead
+                if (this.remoteImageCapture) {
+                    el = createElementLegacy(null, 'ca-remote-image-capture', { name: key, id: key });
+                } else {
+                    el = createElementLegacy(null, 'ca-image-capture', { name: key, id: key });
+                }
+                el.schema = schema.items;
+            }
+
+            return el;
+        }
+
+        /**
+         * @description create select menu
+         * @param {string} type - type of object
+         * @param {string} key - item object key
+         * @param {object} schema - the form schema
+         * @param {string} format - type of item
+         * @returns {object} dom object
+         */
+        createIntSelect(type, key, schema, format) {
+
+            let el = null;
+
+            if (format === 'list' && (type === 'number' || type === 'integer')
+                && Number.isInteger(schema.minimum) && Number.isInteger(schema.maximum)) {
+
+                el = createElementLegacy(null, 'select', { name: key, id: key });
+                createElementLegacy(el, 'option', { value: '' });
+
+                for (let i = schema.minimum; i <= schema.maximum; i++) {
+                    createElementLegacy(el, 'option', { value: i }, i);
+                }
+            }
+
+            return el;
+        }
+
+        /**
+         * @description create a ca-rating dom element
+         * @param {string} type - type of object
+         * @param {string} key - item object key
+         * @param {object} schema - the form schema
+         * @param {string} format - type of item
+         * @returns {object} dom object
+         */
+        createRating(type, key, schema, format) {
+
+            return (format === 'rating') ? createElementLegacy(document.body, 'ca-rating', { name: key, id: key, from: schema.minimum, to: schema.maximum }) : null;
```

Comment:
this should be an if statement - returning nothing will return null anyway, so: 
if(format==='rating') {
  return createElementLegacy(...)
}
does the same thing and doesn't force me to scroll right to see what happens if format isn't rating